lib/ansiblelint/runner.py

```diff
@@ -101,9 +87,25 @@ def run(self) -> List:
                 "Examining %s of type %s",
                 ansiblelint.file_utils.normpath(file['path']),
                 file['type'])
-            matches.extend(self.rules.run(file, tags=set(self.tags),
-                           skip_list=self.skip_list))
+            matches = matches.union(
+                self.rules.run(file, tags=set(self.tags),
+                               skip_list=self.skip_list))
         # update list of checked files
         self.checked_files.update([x['path'] for x in files])
 
-        return sorted(set(matches))
+        return sorted(matches)
+
+    def _yield_matches(self, files: List) -> Generator[MatchError, None, None]:
+        visited: Set = set()
+        while visited != self.playbooks:
+            for arg in self.playbooks - visited:
+                try:
+                    for child in ansiblelint.utils.find_children(arg, self.playbook_dir):
+                        if self.is_excluded(child['path']):
+                            continue
+                        self.playbooks.add((child['path'], child['type']))
+                        files.append(child)
```

Comment:
What exactly can throw an exception here? Just wrap that part with try/except...

Reply:
I kinda like the ability to raise MatchErrors from anywhere but more important: i do not want to change the logic with this change. I can consider changing the try/catch coverage in a follow-up.

I want to keep this change simple, and minimize risk of unexpected side-effects.